src/connector/src/sink/remote.rs

```diff
@@ -0,0 +1,574 @@
+use risingwave_pb::connector_service::sink_config::table_schema::Column;
+use risingwave_pb::connector_service::sink_config::TableSchema;
+use risingwave_pb::connector_service::sink_task::write_batch::json_payload::RowOp;
+use risingwave_pb::connector_service::sink_task::write_batch::{JsonPayload, Payload};
+use risingwave_pb::connector_service::sink_task::{
+    Request as SinkRequest, StartEpoch, StartSink, SyncBatch, WriteBatch,
+};
+use risingwave_pb::connector_service::{SinkConfig, SinkResponse, SinkTask};
+use risingwave_pb::data;
+use risingwave_pb::data::data_type::TypeName;
+use serde_json::Value;
+use serde_json::Value::Number;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+use tokio_stream::wrappers::UnboundedReceiverStream;
+use tokio_stream::StreamExt;
+use tonic::transport::{Channel, Endpoint};
+use tonic::{Request, Status, Streaming};
+
+use crate::sink::{Result, Sink, SinkError};
+
+pub const VALID_REMOTE_SINKS: [&str; 2] = ["jdbc", "file"];
+
+pub fn is_valid_remote_sink(sink_type: String) -> bool {
+    return VALID_REMOTE_SINKS.contains(&sink_type.as_str());
+}
+
+#[derive(Clone, Debug)]
+pub struct RemoteConfig {
+    pub sink_type: String,
+    pub properties: HashMap<String, String>,
+}
+
+impl RemoteConfig {
+    pub fn from_hashmap(values: HashMap<String, String>) -> Result<Self> {
+        let sink_type = values
+            .get("connector")
+            .expect("sink type must be specified")
+            .to_string();
+
+        if !is_valid_remote_sink(sink_type.clone()) {
+            return Err(SinkError::Config(format!("invalid sink type: {sink_type}")));
+        }
+
+        Ok(RemoteConfig {
+            sink_type,
+            properties: values,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub enum RemoteSinkResponseImpl {
+    Grpc(Streaming<SinkResponse>),
+    Receiver(UnboundedReceiver<SinkResponse>),
+}
+
+impl RemoteSinkResponseImpl {
+    pub async fn on_response_ok(&mut self) -> std::result::Result<SinkResponse, SinkError> {
+        return match self {
+            RemoteSinkResponseImpl::Grpc(ref mut response) => response
+                .next()
+                .await
+                .unwrap_or_else(|| Err(Status::cancelled("response stream closed unexpectedly")))
+                .map_err(|e| SinkError::Remote(e.message().to_string())),
+            RemoteSinkResponseImpl::Receiver(ref mut receiver) => receiver
+                .recv()
+                .await
+                .ok_or_else(|| SinkError::Remote("response stream closed unexpectedly".to_string()))
+        };
+    }
+}
+
+#[derive(Debug)]
+pub enum ConnectorClientImpl {
+    Grpc(ConnectorServiceClient<Channel>),
+    Mock,
+}
+
+#[derive(Debug)]
+pub struct RemoteSink {
+    pub sink_type: String,
+    properties: HashMap<String, String>,
+    epoch: Option<u64>,
+    batch_id: u64,
+    schema: Schema,
+    client: ConnectorClientImpl,
```

Comment:
Seems that the `client` is not used anywhere. We may rename it to `_client`. 

Besides, `ConnectorClientImpl` is too verbose. `Option` is enough.